dali/pipeline/pipeline.cc

```diff
@@ -237,14 +238,16 @@ int Pipeline::AddOperator(const OpSpec &spec, const std::string& inst_name) {
 }
 
 int Pipeline::AddOperator(const OpSpec &spec, int logical_id) {
-  return AddOperator(spec, make_string("__", spec.SchemaName(), "_", logical_id), logical_id);
+  std::string name = make_string("__", spec.SchemaName(), "_", logical_id);
+  for (int aux_id = 0; instance_names_.count(name) > 0; aux_id++)
+    name = make_string("__", spec.SchemaName(), "_", logical_id, "_", aux_id);
```

Comment:
Now the `instance_name` must be unique, so a function that doesn't specify one shouldn't possibly fail due to duplicate name - it should generate a unique name.